src/util/GameLogicUtils.cpp

```diff
@@ -96,4 +96,23 @@ namespace spy::util {
 
         return character;
     }
+
+    const util::Point &GameLogicUtils::getRandomFreeNeighbouringField(const gameplay::State &s, const Point &p) {
+        return getRandomNeighbouringField(s, p, [&s](util::Point currentPoint) {
+            // check if point is free -> accessible
+            if (s.getMap().isAccessible(currentPoint)) {
+                return true;
+            }
```

Comment:
```suggestion
            if (!s.getMap().isAccessible(currentPoint)) {
                return false;
            }
```

Sollte das hier nicht negiert sein? Ansonsten returned das Lambda true für alle Felder die betretbar sind oder auf denen kein Charakter steht. Gewollt wäre aus meiner Sicht aber betretbar + kein Charakter drauf

Reply:
durch inline Prüfung der beiden conditions gelöst